clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -250,7 +251,10 @@ private void process(final SyncCommitEvent event) {
 
         try {
             CommitRequestManager manager = requestManagers.commitRequestManager.get();
-            CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> future = manager.commitSync(event.offsets(), event.deadlineMs());
+            CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> future = manager.commitSync(
+                event.offsets().orElseGet(subscriptions::allConsumed),
+                event.deadlineMs()
+            );
```

Comment:
Hi @lianetm 
I believe `commitSync` also needs to get the latest consumed offsets, so I added a `subscriptions::allConsumed` invocation here.